devices/uart-mcxn94x/uart.c

```diff
@@ -0,0 +1,435 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * MCX N94x Serial driver
+ *
+ * Copyright 2020, 2024 Phoenix Systems
+ * Author: Hubert Buczyński, Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <board_config.h>
+
+#include <hal/hal.h>
+#include <lib/lib.h>
+#include <devices/devs.h>
+
+#define BUFFER_SIZE 0x200
+
+typedef struct {
+	volatile u32 *base;
+	unsigned int irq;
+
+	u16 rxFifoSz;
+	u16 txFifoSz;
+
+	u8 rxBuff[BUFFER_SIZE];
+	volatile u16 rxHead;
+	volatile u16 rxTail;
+
+	u8 txBuff[BUFFER_SIZE];
+	volatile u16 txHead;
+	volatile u16 txTail;
+	volatile u8 tFull;
+} uart_t;
+
+
+struct {
+	uart_t uarts[UART_MAX_CNT];
+	int init;
+} uart_common;
+
+
+/* clang-format off */
+enum { veridr = 0, paramr, globalr, pincfgr, baudr, statr, ctrlr, datar, matchr, modirr, fifor, waterr };
+/* clang-format on */
+
+
+static const u32 fifoSzLut[] = { 1, 4, 8, 16, 32, 64, 128, 256 };
+
+
+/* clang-format off */
+const int uartLut[] = {
+	(FLEXCOMM0_SEL == FLEXCOMM_UART) || (FLEXCOMM0_SEL == FLEXCOMM_UARTI2C),
+	(FLEXCOMM1_SEL == FLEXCOMM_UART) || (FLEXCOMM1_SEL == FLEXCOMM_UARTI2C),
+	(FLEXCOMM2_SEL == FLEXCOMM_UART) || (FLEXCOMM2_SEL == FLEXCOMM_UARTI2C),
+	(FLEXCOMM3_SEL == FLEXCOMM_UART) || (FLEXCOMM3_SEL == FLEXCOMM_UARTI2C),
+	(FLEXCOMM4_SEL == FLEXCOMM_UART) || (FLEXCOMM4_SEL == FLEXCOMM_UARTI2C),
+	(FLEXCOMM5_SEL == FLEXCOMM_UART) || (FLEXCOMM5_SEL == FLEXCOMM_UARTI2C),
+	(FLEXCOMM6_SEL == FLEXCOMM_UART) || (FLEXCOMM6_SEL == FLEXCOMM_UARTI2C),
+	(FLEXCOMM7_SEL == FLEXCOMM_UART) || (FLEXCOMM7_SEL == FLEXCOMM_UARTI2C),
+	(FLEXCOMM8_SEL == FLEXCOMM_UART) || (FLEXCOMM8_SEL == FLEXCOMM_UARTI2C),
+	(FLEXCOMM9_SEL == FLEXCOMM_UART) || (FLEXCOMM9_SEL == FLEXCOMM_UARTI2C)
+};
+
+
+static const struct {
+		volatile u32 *base;
```

Comment:
In each member here one tab too many, I think c-f was right :)
```suggestion
	volatile u32 *base;
```